cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define all legal items and values for cylc suite definition files."""
+
+import os
+import re
+
+from metomi.isodatetime.data import Calendar
+
+from cylc.flow import LOG
+from cylc.flow import __version__ as CYLC_VERSION
+from cylc.flow.network.authorisation import Priv
+from cylc.flow.parsec.config import ParsecConfig
+from cylc.flow.parsec.upgrade import upgrader
+from cylc.flow.parsec.validate import (
+    DurationFloat, CylcConfigValidator as VDR, cylc_config_validate)
+from cylc.flow.hostuserutil import get_user_home, is_remote_user
+
+# Nested dict of spec items.
+# Spec value is [value_type, default, allowed_2, allowed_3, ...]
+# where:
+# - value_type: value type (compulsory).
+# - default: the default value (optional).
+# - allowed_2, ...: the only other allowed values of this setting (optional).
+SPEC = {
+    # suite
+    'process pool size': [VDR.V_INTEGER, 4],
+    'process pool timeout': [VDR.V_INTERVAL, DurationFloat(600)],
+    # client
+    'disable interactive command prompts': [VDR.V_BOOLEAN, True],
+    # suite
+    'run directory rolling archive length': [VDR.V_INTEGER, -1],
+    # suite-task communication
+    'task messaging': {
+        'retry interval': [VDR.V_INTERVAL, DurationFloat(5)],
+        'maximum number of tries': [VDR.V_INTEGER, 7],
+        'connection timeout': [VDR.V_INTERVAL, DurationFloat(30)],
+    },
```

Comment:
Is that the whole section or just the `connection timeout` key?

Reply:
I agree with @matthewrmshin on this.  We should document the reasons though so that we don't forget and add retries back in sometime in the future.

Reply:
Can I request that someone with a better understanding than me creates an issue explaining why we are doing this?

Reply:
Yes, the whole `task messaging` section. Task messages from a job fail only when there is a network glitch. It is a very unlikely event. When that happens, the glitch is unlikely to be fixed within 35 seconds (5 * 7) any way! These days, the workflow (suite) server should automatically poll its jobs every now and then, so even if a message is missed, the suite will recover eventually.

Reply:
Any thoughts on how/where to document this - add a comment here? Or put an issue on cylc-admin perhaps?

Reply:
As discussed at desk, please raise a PR against https://cylc.github.io/cylc-admin/rose-suite-run-proposal/cylc-flow-rc.html (or otherwise) to document/propose this change.

Reply:
Messaging retries were removed during ZeroMQ implementation, and the config section here is removed by #3402.  So you can just remove it in this PR @wxtim .  I think we can just document in the User Guide, where task messaging is described, that retries are not attempted (and why that is).

Reply:
I've posted a placeholder issue for the documentation https://github.com/cylc/cylc-doc/issues/87

Reply:
@wxtim - please resolve this conversation after deleting the config section.